gmso/core/topology.py

```diff
@@ -1417,6 +1506,220 @@ def __str__(self):
+                if "." in attr:
+                    try:
+                        attr1, attr2 = attr.split(".")
+                        df[attr + " Atom" + str(site_index)] = list(
+                            _return_float_for_unyt(
+                                getattr(
+                                    getattr(
+                                        connection.connection_members[
+                                            site_index
+                                        ],
+                                        attr1,
+                                    ),
+                                    attr2,
+                                ),
+                                unyts_bool,
+                            )
+                            for connection in getattr(self, parameter)
+                        )
+                    except AttributeError:
+                        raise AttributeError(
+                            "The attribute {} is not in this gmso object".format(
+                                attr
+                            )
+                        )
+                elif attr == "positions" or attr == "position":
+                    df["x Atom" + str(site_index) + " (nm)"] = list(
+                        _return_float_for_unyt(
+                            getattr(
+                                connection.connection_members[site_index],
+                                "position",
+                            )[0],
+                            unyts_bool,
+                        )
+                        for connection in getattr(self, parameter)
+                    )
+                    df["y Atom" + str(site_index) + " (nm)"] = list(
+                        _return_float_for_unyt(
+                            getattr(
+                                connection.connection_members[site_index],
+                                "position",
+                            )[1],
+                            unyts_bool,
+                        )
+                        for connection in getattr(self, parameter)
+                    )
+                    df["z Atom" + str(site_index) + " (nm)"] = list(
+                        _return_float_for_unyt(
+                            getattr(
+                                connection.connection_members[site_index],
+                                "position",
+                            )[2],
+                            unyts_bool,
+                        )
+                        for connection in getattr(self, parameter)
+                    )
+                elif attr == "charge" or attr == "charges":
+                    df["charge Atom" + str(site_index) + " (e)"] = list(
+                        getattr(
+                            connection.connection_members[site_index],
+                            "charge",
+                        ).to_value()
+                        / u.electron_charge.to_value()
```

Comment:
I think we may be able to convert it to e directly with unyt

Reply:
Addressed in d21d2d6b771209a4d2e1b4770267271707f89b17. Had to add a units.py module so we could handle the elementary_charge stuff.